kernel/src/hil/time.rs

```diff
@@ -282,15 +282,6 @@ impl Frequency for Freq16MHz {
     }
 }
 
-/// 24.75MHz `Frequency`
-#[derive(Debug)]
-pub struct Freq2475MHz;
-impl Frequency for Freq2475MHz {
-    fn frequency() -> u32 {
-        24750000
-    }
-}
-
```

Comment:
Why is this being deleted?

Reply:
#1918 introduced `Freq2475MHz`. But after this PR, it's unused. It's replaced by [a frequency value computed at runtime][freq-rt-diff], based on an input clock selection and divider.

I thought this removal, plus the change to compute clock frequency, would do for our current and future goals:

- It let me express the original author's intent -- run the general purpose timers (GPT) on the IPG clock -- while also supporting my goal: run the GPT on the oscillator clock, with root- and peripheral-level clock dividers<sup>*</sup>.
- From the original author's notes, this frequency would eventually change once we configured the ARM clock (which generates the IPG clock):

https://github.com/tock/tock/blob/d695df5223204fe0cf102ccffe097cc208efcc42/chips/imxrt1050/src/gpt1.rs#L247-L253

Although the runtime frequency computation works, I'm looking for other suggestions to solve a perceived _"how to I set a chip's clock frequency, based on the board?"_ problem. It seems I'm the first to propose a `Frequency::frequency()` implementation that doesn't return a constant value, and I'm not sure I want to set that example. Just want to make sure that it's OK to go with this approach.

<sup>* I prefer the 24MHz oscillator clock source since it's not affected by varing processor speeds, it's consistent across all i.MX RT processors, and it's fast enough for me.</sup>

[freq-rt-diff]: https://github.com/tock/tock/pull/2200/files#diff-265db427da6c50259e02f26382d7804a971db41f4b4023ad0c794ce982c80072R318-R333